jsonrpc/src/main.ts

```diff
@@ -840,13 +840,6 @@ function _createMessageConnection(messageReader: MessageReader, messageWriter: M
 				if (progress) {
 					partialResultHandlers.set(String(id), {
 						callback: progress,
-
-						// TODO(nick): This initial array value is a hack because the JSON Patch library that we are using
-						// does not support changing the type of the root document.
-						// See https://github.com/Starcounter-Jack/JSON-Patch/issues/147
-						// Since our first use-case is streaming references, and the type of that response is an array,
-						// initialize the partial result to an empty array.
-						partialValue: [],
```

Comment:
> since that is a native JSON type
Why is that relevant?

The first operation should completely overwrite this value anyway. Undefined seems simpler.